src/SynthGrid.cpp

```diff
@@ -569,6 +579,214 @@ bool SynthGrid::onEvent(SDL_Event& event)
 		{
 			switch (event.key.keysym.sym)
 			{
+
+				case SDLK_RIGHT:
+				{
+					ModularSynth& modularSynth = getModularSynth();
+
+					if (mSelectedModule == -1)
+					{
+						mSelectedModule = 0;
+						mMode = modularSynth.getModule(mSelectedModule) == NULL ? SELECTING_MODULE : IDLE;
+						return true;
+					}
+					if (event.key.keysym.mod & (KMOD_SHIFT) && modularSynth.getModule(mSelectedModule) != NULL && modularSynth.getModule(mSelectedModule)->getNumParams() != 0)
+					{
+						int dialSpeed = 1;
+
+						modularSynth.lock();
+						modularSynth.getModule(mSelectedModule)->onDial(dialSpeed);
+						modularSynth.unlock();
+						setDirty(true);
+					}
+					else
+					{
+						if (mSelectedModule + 1 <= ModularSynth::maxModules - 1 && (mSelectedModule + 1) % 4 != 0)
```

Comment:
What's the reasoning behind this?

Reply:
It's to stop the selection wrapping around to the next line on the grid. No reason it needs to be there, but i just felt it was nicer that it felt like the grid had 'hard' boundaries. If you removed the part beyond the && it will just wrap each edge over to the next line.

Reply:
Ah ok, I get it. :+1:

Reply:
Btw, you will want to check against ```SynthGrid::gridWidth``` instead of 4:

```
if (mSelectedModule < ModularSynth::maxModules - 1 && (mSelectedModule % SynthGrid::gridWidth != 0)) 
```